yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -423,6 +424,7 @@ private[yarn] class YarnAllocator(
     for (completedContainer <- completedContainers) {
       val containerId = completedContainer.getContainerId
       val alreadyReleased = releasedContainers.remove(containerId)
+      var completedContainerReason : String = ""
```

Comment:
I'd prefer declaring this as a `val` inside each conditional, since it's not really used outside of those.